Static/AnalysisStructured.hs

```diff
@@ -686,6 +693,45 @@ anaUnion addSyms lg ln dg nsig name opts eo asps rg = case asps of
           dg3 <- foldM insE dg2 nsigs'
           return (newAsps, nsigs', ns, dg3)
 
+anaIntersect :: Bool -> LogicGraph -> LibEnv -> LibName -> DGraph -> MaybeNode -> NodeName
+  -> HetcatsOpts -> ExpOverrides -> [Annoted SPEC] -> Range
+  -> Result ([Annoted SPEC], [NodeSig], NodeSig, DGraph)
+anaIntersect addSyms lg libEnv ln dg nsig name opts eo asps rg = case asps of
+  [] -> fail "empty intersection"
+  _ -> do
+      let sps = map item asps
+      (sps', nsigs, dg', _) <-
+          let ana (sps1, nsigs, dg', n) sp' = do
+                let n1 = inc n
+                (sp1, nsig', dg1) <-
+                  anaSpec addSyms lg libEnv ln dg' nsig n1 opts eo sp' $ getRange sp'
+                return (sp1 : sps1, nsig' : nsigs, dg1, n1)
```

Comment:
function `ana` should be moved to before the binding `(sps', nsigs, dg', _) <-`, at the same level as `let sps = map item asps`. This will increase readability.